test/extensions/filters/http/cache/http_cache_test.cc

```diff
@@ -170,21 +170,66 @@ TEST_F(LookupRequestTest, NotExpiredViaFallbackheader) {
   EXPECT_EQ(CacheEntryStatus::Ok, lookup_response.cache_entry_status_);
 }
 
-TEST_F(LookupRequestTest, FullRange) {
-  request_headers_.addCopy("Range", "0-99");
+TEST_F(LookupRequestTest, SatisfiableRange) {
+  // add method (GET) and range to headers
+  request_headers_.addCopy(Http::Headers::get().Method.get(),
+                           Http::Headers::get().MethodValues.Get);
+  request_headers_.addCopy(Http::Headers::get().Range.get(), "bytes=1-99,3-,-2");
+
   const LookupRequest lookup_request(request_headers_, current_time_);
+
   const Http::TestResponseHeaderMapImpl response_headers(
       {{"date", formatter_.fromTime(current_time_)},
        {"cache-control", "public, max-age=3600"},
        {"content-length", "4"}});
   const uint64_t content_length = 4;
   const LookupResult lookup_response =
       makeLookupResult(lookup_request, response_headers, content_length);
-  ASSERT_EQ(CacheEntryStatus::Ok, lookup_response.cache_entry_status_);
+  ASSERT_EQ(CacheEntryStatus::SatisfiableRange, lookup_response.cache_entry_status_);
+
   ASSERT_TRUE(lookup_response.headers_);
   EXPECT_THAT(*lookup_response.headers_, Http::IsSupersetOfHeaders(response_headers));
   EXPECT_EQ(lookup_response.content_length_, 4);
-  EXPECT_TRUE(lookup_response.response_ranges_.empty());
+
+  // checks that the ranges have been adjusted to the content's length
+  EXPECT_EQ(lookup_response.response_ranges_.size(), 3);
```

Comment:
Should these be part of the next PR, as they're not part of the parsing changes?

Reply:
The current code already handles the ranges _somewhat_ in `LookupRequest::makeLookupResult`, if they have been parsed in the `LookupRequest` constructor. But since we were not parsing, there were never ranges in `LookupResult` objects. 
Now that we have added the parsing, the objects created have ranges, so I adjusted these tests accordingly.

The handling that needs to be implemented is in `CacheFilter::onHeaders`, as we are not breaking down the body according to the ranges we have.